src/main/java/com/three_tech_solutions/slot_app/components/payment_processors/implementations/PrincipioDeMesPaymentProcessor.java

```diff
@@ -0,0 +1,21 @@
+package com.three_tech_solutions.slot_app.components.payment_processors.implementations;
+
+import com.three_tech_solutions.slot_app.components.payment_processors.PaymentProcessor;
+import com.three_tech_solutions.slot_app.data.enums.PlanType;
+import com.three_tech_solutions.slot_app.data.models.Student;
+import org.springframework.stereotype.Component;
+import java.time.LocalDateTime;
+
+@Component
+public class PrincipioDeMesPaymentProcessor extends PaymentProcessor {
+
+    @Override
+    public PlanType getCurrentPlan() {
+        return PlanType.PRINCIPIO_DE_MES;
+    }
+
+    @Override
+    public LocalDateTime getExpirationDate(Student student) {
+        return LocalDateTime.now().withDayOfMonth(10);
```

Comment:
Esto es una boludés, pero que a veces mejora la legibilidad/claridad del código.

En vez de simplementar decir que el día del mes es 10, podemos crear una constante "PRINCIPIO_DE_MES_EXPIRATION_DATE" que tenga el valor 10, por lo tanto el codigo nos quedaría así

Creamos esta constante en la clase:

`private final int PRINCIPIO_DE_MES_EXPIRATION_DATE = 10;`

y acá cambiamos el 10 por esta constante

```suggestion
        return LocalDateTime.now().withDayOfMonth(PRINCIPIO_DE_MES_EXPIRATION_DATE);
```
De manera que sabemos qué es lo que significa ese 10 simplemente con leerlo